### DISCUSSION TITLE
Configuration options menu

### DISCUSSION THREAD

**Discussion by LwsBtlr** [ExistentialAudio/BlackHole#7] [Feature Requests]
[Discussion] Will there be any configuration options for this, or is it going to remain as a simple pass through tool like it is now?

I’d like to be able to, for example, pass a flag to start a new file ever 5 minutes (or every chapter if the source has chapters, or pass a list of time codes). Not sure if this is something that might be done or something totally out of the scope of the project (I suspect the latter).

Is It possible to integrate it into a command line pipe chain? So I could direct my audio app to play to blackhole and tell… I don’t know, FFmpeg to listen to blackhole? Or does it make more sense just to use black hole to get the audio and then reprocess the audio after the fact?

**Comment by devinroth** [comment]
I have plans to implement some configuration options. I want to the ability to change the number of channels with a simple menu option along with changing the clock source and output/input mapping. 

For the other ideas that you have I think they are beyond the scope of what I intend with his. All of those make more sense to me to be done in a DAW. At least for now I intend to keep things simple but if enough people are asking for specific functions I may roll them in.

**Comment by Andertve** [comment]
Would be great to have a configuration/menu to increase the number of channels without having to make a custom build.

**Comment by devinroth** [comment]
Agree. It’s in the works.

**Reply by gianpaj** [reply]
How were you thinking about developing this? I'm a React Native and a new iOS developer. I can give it a go if you have any suggestions + links on how to do this.
Including how to trigger the events that will change the output/input mapping.
Thanks

**Comment by QuentinBrown** [comment]
> I have plans to implement some configuration options. I want to the ability to change the number of channels with a simple menu option along with changing the clock source and output/input mapping.
> 
> For the other ideas that you have I think they are beyond the scope of what I intend with his. All of those make more sense to me to be done in a DAW. At least for now I intend to keep things simple but if enough people are asking for specific functions I may roll them in.

And here's one additional request that I hope you might consider - the ability to launch and possibly manage additional instances of blackhole from the configuration options menu. This way, when combined with the ability to change the number of channels for each instance running, you wouldn't need to build and install multiple fixed instances to cope with the need to send and receive between multiple non-channel aware apps like Skype or zoom. Hopefully anyway. I'm not sure if macOS architecture allows things to work this way on the fly or if it has to actually have a driver instance hard installed.

For now, I'm off to try and follow the instructions for building my own extra 2 channel instances with different UIDs. Thanks for putting those up!

**Comment by ivanvfer** [comment]
It would be great to support additional Blackhole 2ch instances. This would provide more flexibility in routing traffic among multiple non-channel aware apps. 

A way to support this could be to have a Blackhole 2ch installer that uses a random UID/name suffix. This way, the same installer could be used to create multiple BlackHole 2ch instances, which would run as independent versions.

**Reply by GrampaPaul** [reply]
two channels is ideal for us non complex users

**Comment by freakm941** [comment]
i want this feature

**Reply by fuguesoft** [reply]
I'm using this with an aggregate device in order to capture desktop audio, send that to Ableton for mixing and send all that into Discord.

The very teeny tiny, super small, mite-sized inconvenience I'm having is that I have to change my System Audio from the aggregate device to Black hole 16ch to get my desktop audio to go into Ableton.

So I either have to: 
1) Have Ableton always open to hear Desktop Audio _or_ 
2) Hear my Desktop Audio through my interface and main speakers but not be able to route it into Discord

I found that, by default, if an application doesn't provide the option to change which channel of BH to use, it will use 1 and 2.

Because I have an audio interface with 6in/6out, BH 1/2 becomes 7/8 in my aggregate device which is why I'm having this inconvenience

So finally to my question (lol)

Will the "custom configurations" feature allow me to circumvent this? Or is this more of a macOS issue? Also should I post this elsewhere instead?

Thank you

- L33t

**Comment by devinroth** [comment]
Multi-output?

Devin Roth

**Reply by fuguesoft** [reply]
Multi output freezes youtube and and some other players as they don't know what to do with it

Also because I'm routing busses from my DAW, the aggregate option works the best as of now

**Comment by devinroth** [comment]
Known issue. Read some of the other issues and faq to solve.

Devin Roth

**Reply by fuguesoft** [reply]
Talking about this?

> Why is audio glitching after X minutes when using a multi-output or

>  an aggregate?
> 
> You need to enable drift correction for all devices except the clock source device.

Because that doesn't solve the issue

**Reply by fuguesoft** [reply]
Found the solution, however now I'm unable to mix/process the desktop audio before it hits discord

So it looks like my previous solution would be just something I have to deal with

**Comment by devinroth** [comment]
Yes.

But I noticed a bug that may be causing that too.

Try this beta and see if it makes any difference.

https://www.dropbox.com/s/oog8fln3ihsmnmt/BlackHole2ch.v0.2.10.beta.pkg.zip?dl=0

Devin Roth

**Reply by fuguesoft** [reply]
Can I use the beta with the current 16ch version I have installed?
E: Found the Advanced Customization page

**Comment by devinroth** [comment]
Maybe? I think the beta is 2ch so it should work.

Devin Roth

**Reply by fuguesoft** [reply]
Oh no it doesn't make a difference
I've set Discord's input to be the 2ch beta and the device configured like this:
![image](https://user-images.githubusercontent.com/31870368/123554658-9b698b00-d746-11eb-8c3f-637556d98835.png)

I'm assuming, that because the 2ch is at the bottom, those will be the last inputs to show in my DAW

I'm not routing it through Live to those last inputs so it is either:
1) Bypassing Live completely OR
2) positioning itself at the top of the input stack and pushing the 16ch down the list


E1: I'm noticing that setting Discord's input to the 16ch, give the same result however...

Audio going in to Discord but not being routed by Live

E2: But I think that's how the multi output works

**Reply by fuguesoft** [reply]
Yeah so I'm back to my original thing
If this feature would fix that it definitely has my vote

**Reply by fuguesoft** [reply]
Got it working with the multi-output and vb cable!
Thanks for the help!
